src/fu-engine.c

```diff
@@ -256,6 +257,12 @@ fu_engine_set_status(FuEngine *self, FwupdStatus status)
 static void
 fu_engine_generic_notify_cb(FuDevice *device, GParamSpec *pspec, FuEngine *self)
 {
+	if (fu_idle_get_status(self->idle) == FWUPD_STATUS_DEVICE_BUSY &&
```

Comment:
Busy should be enough for this case, but shouldn't more be covered? What if you really unplugged a power cable not part of a dock during update? It would still trip the problem I think.

Reply:
I think I've confused you re-using the FwupdStatus as the status of the idle handler -- it was lazyness on my part I guess. I've replaced it with a custom enum to make the state machine and possible states a bit clearer. Is that better?

Reply:
Yeah that's clearer.